lib/iris/tests/unit/fileformats/netcdf/load_cube/load_cube__activate/test__grid_mappings.py

```diff
@@ -0,0 +1,866 @@
+        to support different grid-mapping types (for example).
+
+        """
+        # The grid-mapping options are standard-latlon, rotated, or non-latlon.
+        # This affects names+units of the X and Y coords.
+        # We don't have an option to *not* include a grid-mapping variable, but
+        # we can mimic a missing grid-mapping by changing the varname from that
+        # which the data-variable refers to, with "gridmapvar_name=xxx".
+        # Likewise, an invalid (unrecognised) grid-mapping can be mimicked by
+        # selecting an unkown 'grid_mapping_name' property, with
+        # "gridmapvar_mappropertyname=xxx".
+        if mapping_type_name is None:
+            # Default grid-mapping and coords are standard lat-lon.
+            mapping_type_name = hh.CF_GRID_MAPPING_LAT_LON
+            xco_name_default = hh.CF_VALUE_STD_NAME_LON
+            yco_name_default = hh.CF_VALUE_STD_NAME_LAT
+            xco_units_default = "degrees_east"
+            # Special kwarg overrides some of the values.
+            if latitude_units is None:
+                yco_units_default = "degrees_north"
+            else:
+                # Override the latitude units (to invalidate).
+                yco_units_default = latitude_units
+
+        elif mapping_type_name == hh.CF_GRID_MAPPING_ROTATED_LAT_LON:
+            # Rotated lat-lon coordinates.
+            xco_name_default = hh.CF_VALUE_STD_NAME_GRID_LON
+            yco_name_default = hh.CF_VALUE_STD_NAME_GRID_LAT
+            xco_units_default = "degrees"
+            yco_units_default = "degrees"
+
+        else:
+            # General non-latlon coordinates
+            # Exactly which depends on the grid_mapping name.
+            xco_name_default = hh.CF_VALUE_STD_NAME_PROJ_X
+            yco_name_default = hh.CF_VALUE_STD_NAME_PROJ_Y
+            xco_units_default = "m"
+            yco_units_default = "m"
+
+        # Options can override coord (standard) names and units.
+        if xco_name is None:
+            xco_name = xco_name_default
+        if yco_name is None:
+            yco_name = yco_name_default
+        if xco_units is None:
+            xco_units = xco_units_default
+        if yco_units is None:
+            yco_units = yco_units_default
+
+        phenom_auxcoord_names = []
+        if xco_is_dim:
+            # xdim has same name as xco, making xco a dim-coord
+            xdim_name = "xco"
+        else:
+            # use alternate dim-name, and put xco on the 'coords' list
+            xdim_name = "xdim_altname"
+            phenom_auxcoord_names.append("xco")
+        if yco_is_dim:
+            # ydim has same name as yco, making yco a dim-coord
+            ydim_name = "yco"  # This makes the X coord a dim-coord
+        else:
+            # use alternate dim-name, and put yco on the 'coords' list
+            ydim_name = "ydim_altname"  # This makes the X coord a dim-coord
```

Comment:
Same as above, I don't see how this would affect the X coord.

Reply:
:+1: